### PR TITLE
consumer: fix panic in nextLookupdEndpoint

### PR DESCRIPTION
**Pull request by martin-sucha**
The code can panic if invalid lookupd address is provided:

    panic: parse "http:// myhostname:4161": invalid character " " in host name

This is caused different validation of the input when configured
in ConnectToNSQLookupd and when used in nextLookupdEndpoint.

Moving preparation of the URL to the beginning so that we don't
need to do any input validation in nextLookupEndpoint.

Fixes https://github.com/nsqio/go-nsq/issues/316

### CODE REVIEW

**Comment by martin-sucha** [comment]
Note that I have not tested the code yet as I don't have a nsq instance on the machine I'm currently on.

**Comment by mreiferson** [comment]
I'm confused, it looks like `validatedLookupAddr` already returned an error from `url.Parse`, perhaps in your test case you're not checking the return value from `ConnectToNSQLookupd`?

**Comment by martin-sucha** [comment]
If you use `" myhostname:4161"` as the input value then `validatedLookupAddr` does not call `url.Parse` at all because the string does not contain a slash.

**Comment by martin-sucha** [comment]
I've updated the commit message so that it is clearer.

File: consumer.go [hunk 1]
```diff
@@ -126,7 +126,7 @@ type Consumer struct {
 
 	// used at connection close to force a possible reconnect
 	lookupdRecheckChan chan int
-	lookupdHTTPAddrs   []string
+	lookupdHTTPAddrs   []lookupdHTTPAddr
```
Comment (mreiferson) [inline consumer.go:129]: I don't see any benefit of introducing a type here, let's keep this a `string` and then a few of the other changes here won't be necessary.

**Review by mreiferson — CHANGES_REQUESTED** [review]
Thanks @martin-sucha, I think we can simplify this a bit so the diff isn't as noisy.

File: consumer.go (hunk 1 above)
Comment (martin-sucha) [reply consumer.go]: Ok, I removed the type.

**Pull request merged**